src/coreclr/jit/hwintrinsic.h

```diff
@@ -70,175 +70,176 @@ enum HWIntrinsicCategory : uint8_t
 #else
 #error Unsupported platform
 #endif
+
 enum HWIntrinsicFlag : unsigned int
 {
     HW_Flag_NoFlag = 0,
 
     // Commutative
     // - if a binary-op intrinsic is commutative (e.g., Add, Multiply), its op1 can be contained
-    HW_Flag_Commutative = 0x1,
+    HW_Flag_Commutative = (1 << 1),
```

Comment:
This enum muddled - it has common entries, then target specific, then more common entries (eg HW_Flag_SpecialSideEffect_Barrier).

I've reordered it, and then used defines for the target specific entries to make it easier to add common entries later. More common entires can be added after `HW_Flag_CanBenefitFromConstantProp` and only `HW_Flag_PlatformBase` needs updating